public/components/common/custom-search-bar/components/multi-select.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * Wazuh app - React component Multi-select
+ * Copyright (C) 2015-2021 Wazuh, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Find more information about this on the LICENSE file.
+ */
+
+import React, { useEffect, useState } from 'react';
+import {
+  EuiFieldSearch,
+  EuiFilterButton,
+  EuiFilterGroup,
+  EuiFilterSelectItem,
+  EuiIcon,
+  EuiLoadingChart,
+  EuiPopover,
+  EuiPopoverTitle,
+  EuiSpacer,
+  FilterChecked,
+} from '@elastic/eui';
+import { IValueSuggestion, useValueSuggestion } from '../../hooks';
+
+const ON = 'on';
+const OFF = 'off';
+
+export const MultiSelect = ({ item, onChange, selectedOptions, onRemove }) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
+  const { suggestedValues, isLoading, setQuery }: IValueSuggestion = useValueSuggestion(
+    item.key,
+    item?.options
+  );
+  const [items, setItems] = useState<
+    { key: any; label: any; value: any; checked: FilterChecked }[]
+  >([]);
+  const [activeFilters, setActiveFilters] = useState<number>(0);
+
+  useEffect(() => {
+    if (!isLoading) {
+      setItems(
+        suggestedValues
+          .map((value, key) => ({
+            key: key,
+            label: value,
+            value: item.key,
+            filterByKey: item.filterByKey,
+            checked: OFF as FilterChecked,
+          }))
+          .sort((a, b) => a.label - b.label)
+      );
+    }
+  }, [suggestedValues, isLoading, setQuery]);
+
+  useEffect(() => {
+    items.forEach((item) => {
+      item.checked = selectedOptions.find((element) => element.label === filterBy(item)) ? ON : OFF;
```

Comment:
`items` is an state object returned by `useState` and should be immutable. Switching the `checked` value directly goes against this as it doesn't make React aware of the changes to the object and may cause unexpected issues.